fixtures/coverall/src/coverall.udl

```diff
@@ -47,6 +50,10 @@ interface ComplexError {
 interface Coveralls {
     constructor(string name);
 
+    // There's one "global" coverall - this creates or returns it via an Arc<>
+    [Name="global_new", Self=ByArc]
+    constructor();
```

Comment:
If I'm reading this correctly:

```swift
let c1 = Coveralls.globalNew()
let c2 = Coveralls.globalNew()

assert c1 != c2 // fail
```

If we named this `get_instance`, we can see that this isn't a constructor at all.

Have I got that right? Was this your intent?

Reply:
The intention is that `assert c1 == c2` would pass. As part of that process, a `Coveralls` object was constructed.

I'm trying to wrap the pattern at https://github.com/mozilla/application-services/blob/a3d01428113d4a237c4a76e7095efad9c13fe958/components/places/src/api/places_api.rs#L90 and exposed to Kotlin at https://github.com/mozilla/application-services/blob/a3d01428113d4a237c4a76e7095efad9c13fe958/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt#L37. I think it's a perfectly reasonable pattern - you might have chosen to use a different pattern, but I think it would be very brave to say that the places_api pattern is objectively wrong. I don't think UniFFI should be so opinionated that it refuses to support that pattern.

Reply:
Gosh.